language/tools/move-cli/src/main.rs

```diff
@@ -0,0 +1,440 @@
+    move_lib: String,
+    /// If false, compile `move_data` against the Move stdlib and include the stdlib bytecodes in
+    /// `move run`.
+    #[structopt(name = "no-use-stdlib", long = "no-use-stdlib")]
+    no_use_stdlib: bool,
+    // Command to be run.
+    #[structopt(subcommand)]
+    cmd: Command,
+}
+
+#[derive(StructOpt)]
+enum Command {
+    /// Compile the modules in `move_data` and the given script (if any).
+    #[structopt(name = "compile")]
+    Compile {
+        /// Path to script to compile.
+        #[structopt(name = "script")]
+        script_file: Option<String>,
+        /// If true, commit the bytecodes for the compiled modules to disk.
+        #[structopt(long = "commit", short = "c")]
+        commit: bool,
+    },
+    /// Compile/run a Move script that reads/writes resources stored on disk in `move_data`. This
+    /// command compiles each each module stored in `move_lib` and loads it into the VM before
+    /// running the script.
+    #[structopt(name = "run")]
+    Run {
+        /// Path to script to compile and run.
+        #[structopt(name = "script")]
+        script_file: String,
+        /// Possibly-empty list of signers for the current transaction (e.g., `account` in
+        /// `main(&account: signer)`). Must match the number of signers expected by `script_file`.
+        #[structopt(long = "signers")]
+        signers: Vec<String>,
+        /// Possibly-empty list of arguments passed to the transaction (e.g., `i` in
+        /// `main(i: u64)`). Must match the arguments types expected by `script_file`.
+        #[structopt(long = "args", parse(try_from_str = parser::parse_transaction_argument))]
+        args: Vec<TransactionArgument>,
+        /// Possibly-empty list of type arguments passed to the transaction (e.g., `T` in
+        /// `main<T>()`). Must match the type arguments kinds expected by `script_file`.
+        #[structopt(long = "type-args", parse(try_from_str = parser::parse_type_tag))]
+        type_args: Vec<TypeTag>,
+        /// If true, commit the effects of executing `script_file` (i.e., published, updated, and
+        /// deleted resources) to disk.
+        #[structopt(long = "commit", short = "c")]
+        commit: bool,
+    },
+    /// View Move resources and modules stored on disk
+    #[structopt(name = "view")]
+    View {
+        /// Path to a resource or module stored on disk.
+        #[structopt(name = "file")]
+        file: String,
+    },
+    /// Delete all modules and resources stored on disk under `move_data`. Does *not* delete
+    /// anything in `move_lib`.
+    Clean {},
+}
+
+/// Create a directory at ./`dir_name` if one does not already exist
+fn maybe_create_dir(dir_name: &str) -> Result<&Path> {
+    let dir = Path::new(dir_name);
+    if !dir.exists() {
+        fs::create_dir(dir)?
+    }
+    Ok(dir)
+}
+
+/// Compile the user modules in `move_lib_directory` and the script in `script_file`
```

Comment:
It seems like this is an all at once thing for modules? Am I understanding that correctly?

I have some ideas on how to improve this. And talking to Victor earlier about testing, I got some ideas of what to do to improve both of these use cases (hopefully) 

But that discussion is not needed for this PR

Reply:
Yes, that's correct. It's super naive today--it compiles and publishes everything in `move_lib` even if the bytecodes have already been published to `move_data`.